feat(tag, closing-button): Component Tag and ClosingButton #166 #203

airikej opened the pull request.

kasparsuvi1 commented on libs/react-components/tedi/src/components/closing-button/closing-button.stories.tsx:
```diff
@@ -0,0 +1,90 @@
+  render: SizeTemplate,
+};
+
+export const IconSize: Story = {
```
```suggestion
export const Hover: Story = {
```

kasparsuvi1 commented on libs/react-components/tedi/src/components/closing-button/closing-button.stories.tsx:
```diff
@@ -0,0 +1,90 @@
+  );
+};
+
+const IconSizeTemplate: StoryFn = () => {
```
Do we need 2 separate templates for it? Hover state can be passed as prop, everything else is shared?

kasparsuvi1 commented on libs/react-components/tedi/src/components/closing-button/closing-button.stories.tsx:
```diff
@@ -0,0 +1,90 @@
+};
+
+const sizeArray: ClosingButtonProps['size'][] = ['small', 'medium', 'large'];
+const iconSizeMap = {
```
Is not used by component and i think it should not be demo'd like that.

kasparsuvi1 commented on libs/react-components/tedi/src/components/closing-button/closing-button.stories.tsx:
```diff
@@ -0,0 +1,90 @@
+  title: 'Tedi-ready/Components/ClosingButton',
+  parameters: {
+    status: {
+      type: [{ name: 'internalComponent' }],
```
STAR uses current CloseButton also, to achive closable Tag's.
Probaly we can use New StatusBadge and old Tags togheter in UI. (Because it's the same color and size, but just a note.)

airikej replied:
So you would like ClosingButton to be exported from library as well?

kasparsuvi1 replied:
For STAR it's not needed, but maybe somebody else also uses it? so it would be a breaking change.

kasparsuvi1 commented on libs/react-components/tedi/src/components/tag/tag.module.scss:
```diff
@@ -0,0 +1,49 @@
+    }
+  }
+
+  &__close {
```
Where is this class used?

kasparsuvi1 commented on libs/react-components/tedi/src/components/tag/tag.stories.tsx:
```diff
@@ -0,0 +1,81 @@
+      type: [{ name: 'breakpointSupport', url: '?path=/docs/helpers-usebreakpointprops--usebreakpointprops' }],
+    },
+    controls: {
+      exclude: ['sm', 'md', 'lg', 'xl', 'xxl'],
```
👍

kasparsuvi1 commented on libs/react-components/tedi/src/components/tag/tag.tsx:
```diff
@@ -0,0 +1,64 @@
+   * Function to be called when the close button is clicked.
+   * If provided, a close button will be rendered inside the Tag.
+   */
+  onClose?: MouseEventHandler<HTMLButtonElement>;
```
Doesn't need to be in breakpointProps, no need to call different onClose trigger on different breakpoints

kasparsuvi1 commented on libs/react-components/tedi/src/components/tag/tag.tsx:
```diff
@@ -0,0 +1,64 @@
+   * Determines whether the Tag is in a loading state
+   * @default false
+   */
+  isLoading?: boolean;
```
Same isLoading state doesn't different by breakpoint.

kasparsuvi1 commented on libs/react-components/tedi/src/components/tag/tag.tsx:
```diff
@@ -0,0 +1,64 @@
+  /**
+   * Additional classes to apply custom styles to the Tag.
+   */
+  className?: string;
```
ClassName can be needed more for different breakpoints i assume.

kasparsuvi1 commented on libs/react-components/tedi/src/components/tag/tag.tsx:
```diff
@@ -0,0 +1,64 @@
+    ...rest
+  } = getCurrentBreakpointProps<TagProps>(props);
+
+  const tagBEM = cn(styles['tedi-tag'], color && styles[`tedi-tag--color-${color}`], className);
```
Btw, why we use `tedi-tag` not just `tag`? Modular css already gives us special classnames. It just makes CSS longer and harder to read DOM

airikej replied:
This was to minimise the overriding styles of the Community Tag component, since we won't be removing it right away

kasparsuvi1 replied:
But there is no need really?

airikej replied:
Theoretically no, but TEDI team agreed to have tedi classes

kasparsuvi1 commented on libs/react-components/tedi/src/components/closing-button/closing-button.tsx:
```diff
@@ -0,0 +1,60 @@
+import cn from 'classnames';
```
I think we should also use this in Modal, currently there is custom Button

airikej replied:
Agreed, I looked into Figma, Modal component was actually ready for dev so It would make sense to replace it during the Modal development, I would rather not break anything in Community's modal in this PR :D

kasparsuvi1 commented:
Also its not good practice to publish 3 different things in one PR.
Currently this PR adds: ClosingButton, Tag and also fix bugs with variables. Very hard to see the difference in Changelog later when updating library.

airikej commented:
@kasparsuvi1 I've replaced ClosingButtons and added Tag to FileUpload, please check these:
https://github.tehik.ee/tedi-design-system/feat/TEDI-366-Component-Tag/?path=/docs/community-form-fileupload--docs
https://github.tehik.ee/tedi-design-system/feat/TEDI-366-Component-Tag/?path=/docs/community-modal--docs
and I found this thing as well, where I replaced ClosingButton:
https://github.tehik.ee/tedi-design-system/feat/TEDI-366-Component-Tag/?path=/docs/community-layout-header-headermodal--docs

kasparsuvi1 commented on libs/react-components/community/src/components/modal/modal-closer.tsx:
Seda ei tohiks kindlasti ära kustutada, see on helper component, mida saab wrappida stiilis:
```
<ModalCloser>
 <Button ../>
```
Ja seda kasutatakse pea kõigis rakendustes ala "Tühista", "Sulge"  jne nuppude ümber koodi lihtsustamiseks. (Pole visuaalne komponent.)

kasparsuvi1 commented on libs/react-components/community/src/components/layout/header/header/header.module.scss:
```diff
@@ -68,3 +68,11 @@ $header-min-height: 3.5rem;
+
+.header-modal {
+  &__close {
+    --general-icon-primary: var(--primary-600);
```
ka Hoveril peaks colori üle kirjutama.
(Kuigi ma eelistaks mitte sellist komponentide stiilide üle kirjutamist (kergem asju ära lõhkuda). See primary color võiks pigem olla CloseButtoni propidega saavutatav.)

airikej replied:
Kas siin ei võiks üldsegi tavaline ClosingButton stiiliga olla? Tundub nii niche case, et tegelt ei tahaks üldse selle stiili muuta

kasparsuvi1 replied:
TP disaineri poolt okei. Ma ei tea ka, et teistes rakendustes mobiili versiooni headeris kasutatakse. 👍

kasparsuvi1 reviewed (commented):
Jah, ma täpselt ei saanud aru miks ModalClose komponent oli vaja nüüd eemaldada? See ei ole otseselt visuaalne komponent vaid helper, mis Modalit kasutada aitab.

kasparsuvi1 commented on libs/react-components/community/src/components/modal/modal.tsx:
```diff
@@ -86,6 +86,7 @@ export const Modal = (props: ModalProps): JSX.Element | null => {
   const labelId = props['aria-labelledby'];
   const descriptionId = props['aria-describedby'];
   const { isOpen, floating, getFloatingProps, context, isDismissable } = React.useContext(ModalContext);
+  const modalContentRef = React.useRef<HTMLDivElement | null>(null);
```
See on üleliigselt jäänud?

airikej replied:
Mõtlesin, et on kahe silma vahele jäänud tõesti, aga see aitab selle vastu, et close button poleks modalit avades kohe focus state's
